### ISSUE TITLE
Highlight the replacer wagon when selecting the replacee

### ISSUE THREAD

**Issue by DorpsGek** [OpenTTD/OpenTTD#4549]
**_Zekonnd_ opened the ticket and wrote:**

> Windows XP
> Using the "replace train" window for replacing wagons.
> 
> Steps: 
> 1. Select a wagon A (savegame: Gondola) in LH column as replacee.
> 2. Select any wagon, except the first one (wagon C) in RH column (savegame: Goods Van) as replacer wagon
> 3. Press Start replacing vehicles.
> 4. Select another wagon B in LH side (savegame: stake wagon)
> 5. Select another wagon D (savegame: low side wagon) in RH side, different from wagon C! 
> 6. Select wagon A once more in LH side.
> 
> Expected: In the RH column wagon C is higlighted.
> 
> Instead: Wagon D in RH column is still selected.
> 
> Notes: 
> \- The correct name of wagon C is displayed in center info area after step 6. 
> \- I did not check for "scroll into view" issues.
> \- I cannot remember the version, but it worked before (0.7 i think).
> \- Using dbsetxl new grf
> \- Savegame just created one depot, one train and two wagons, then reproduced behaviour
> \- Will also happen, when closing and reopen replace train window instead of steps 4+5 (first wagon stays selected in RH side)
> 
> Best regards and many thanks for doing all OTTD programming and releasing work!
> Zekonnd
> 

**Attachments**
- [Bugreport-Wagonreplacement.sav](https://bugs.openttd.org/task/4549/getfile/7398/Bugreport-Wagonreplacement.sav) (279.02 KiB)
- [dbsetxlw.grf](https://bugs.openttd.org/task/4549/getfile/7399/dbsetxlw.grf) (902.32 KiB)

**Reported version**: _1.1.0-RC2_
**Operating system**: _Windows_

---

**Comment by DorpsGek** [comment]
**_Rubidium_ wrote:**

> What's wrong with keeping wagon D selected?
> 
> Given it didn't even work like the way you expect it to behave in 0.5, 0.6, 0.7 and 1.0, I think this isn't a regression.

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4549#comment9763_

**Comment by DorpsGek** [comment]
**_Zekonnd_ wrote:**

> My problem was, that at first i actually did not notice, that i assigned a wrong wagon C to A. Because I just did not look at the center info area, where the name of wagon C is displayed, because i was used to look in the RH column.
> 
> The advantage of the expected behaviour is, in my opinion, that you also can compare wagon properties (A+C) with a single click on A, specifically capacity and speed.
> 
> I am pretty sure that it did work in 0.7 as i described, at least with trains. If this is not the case, or if this is not enough for beeing a regression issue, then please close the issue, or postpone it as an enhancement! Sorry, i was not able obtain 0.7 to check this myself.
> 
> The issue described also refers to train replacement BTW.

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4549#comment9764_

**Comment by DorpsGek** [comment]
**_planetmaker_ wrote:**

> Get every released binary here: http://binaries.openttd.org/releases/index.html

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4549#comment9765_

**Comment by DorpsGek** [comment]
**_andythenorth_ closed the ticket.**

**Reason for closing: _Works for me_**

> Tested around r27892, auto-replace UI works as expected for me.  No bugs.

---
###### _This comment was imported from FlySpray: https://bugs.openttd.org/task/4549_
